exec/java-exec/src/main/java/org/apache/drill/exec/ops/OperatorContextImpl.java

```diff
@@ -130,6 +134,60 @@ public OperatorStats getStats() {
     return stats;
   }
 
+
```

Comment:
Can we move create a pool for these delegate threads? The default can be empty but we'll expand as needed. We also need to think about interruptions and errors. It might be better if the actual method interface is Future<Result> and we allow the implementor pick when to block. Guava has a abstract checked future that would probably solve most of these needs.